packages/vite/src/build.ts

```diff
@@ -0,0 +1,99 @@
+import { join } from 'path/posix';
+import { createHash } from 'crypto';
+import { fork } from 'child_process';
+import { existsSync, readdirSync, readFileSync, writeFileSync } from 'fs';
+import type { Plugin } from 'vite';
+
+const cwd = process.cwd();
+const embroiderDir = join(cwd, 'node_modules', '.embroider');
+const cacheKeyPath = join(embroiderDir, 'cache-key.json');
+
+export const lockFiles = ['pnpm-lock.yaml', 'package-lock.json', 'yarn.lock', 'package.json'];
+
+function getCacheKey(file: string) {
+  if (existsSync(cacheKeyPath)) {
+    return JSON.parse(readFileSync(cacheKeyPath).toString())[file];
+  }
+  return null;
+}
+
+function updateCacheKey(file: string, key: string | null) {
+  let json: Record<string, string | null> = {};
+  if (existsSync(cacheKeyPath)) {
+    json = JSON.parse(readFileSync(cacheKeyPath).toString());
+  }
+  json[file] = key;
+  writeFileSync(cacheKeyPath, JSON.stringify(json));
+}
+
+function computeCacheKeyForFile(file: string) {
+  if (existsSync(file)) {
+    const fileBuffer = readFileSync(file);
+    const hashSum = createHash('sha256');
+    hashSum.update(fileBuffer);
+
+    return hashSum.digest('hex');
+  }
+  return null;
+}
+
+export function emberBuild(mode: string): Promise<void> {
+  if (mode === 'build') {
+    return new Promise((resolve, reject) => {
+      const child = fork('./node_modules/ember-cli/bin/ember', ['build', '--production'], { silent: true });
+      child.on('exit', code => (code === 0 ? resolve() : reject()));
+    });
+  }
+  return new Promise((resolve, reject) => {
+    const child = fork('./node_modules/ember-cli/bin/ember', ['build', '--watch'], { silent: true });
+    child.on('exit', code => (code === 0 ? resolve() : reject(new Error('ember build --watch failed'))));
+    child.on('spawn', () => {
+      child.stderr?.on('data', data => {
+        console.error(data.toString());
+      });
+      child.stdout!.on('data', data => {
+        console.log(data.toString());
+        if (data.toString().includes('Build successful')) {
+          resolve();
+        }
+      });
+    });
+  });
+}
+
+export async function needForcedRebundle(path: string | null | undefined): Promise<boolean> {
```

Comment:
can we not add this? is this working around a bug (possibly something not being watched correctly in broccoli?) or is this actually needed for a specific reason?

Reply:
Its required if deps changed. Otherwise you will get issues with outated deps, embroider will import addons directly instead of rewritten packages

Reply:
The outdated dep is common. And some who have experience with vite already now that they need to start vite with --force. But there are akso other issues with the embroider rewritten packages if they are outdated

Reply:
> Its required if deps changed

But ember-cli has never supported changing deps during a running server 🤔 this seems like a lot of complexity to add for a new functionality like this

Reply:
well, it can be done in another pr if the need arises. removed

Reply:
Counterpoint, everyone i've talked to outside of (and new to) the ember community expects this behavior